crates/compression/src/registry.rs

```diff
@@ -0,0 +1,109 @@
+use crate::ports::{
+    EvictorDb,
+    TemporalRegistry,
+};
+use fuel_core_types::{
+    fuel_compression::RegistryKey,
+    fuel_tx::{
+        input::PredicateCode,
+        Address,
+        AssetId,
+        ContractId,
+        ScriptCode,
+    },
+    tai64::Tai64,
+};
+
+macro_rules! tables {
+    ($($ident:ty: $type:ty),*) => { paste::paste! {
+        #[doc = "RegistryKey namespaces"]
+        #[derive(Debug, Clone, Copy, PartialEq, Eq, Hash, serde::Serialize, serde::Deserialize, strum_macros::EnumCount)]
+        pub enum RegistryKeyspace {
+            $(
+                [<$type>],
+            )*
+        }
+
+        #[doc = "A value for each keyspace"]
+        #[derive(Debug, Clone, PartialEq, Eq, Default)]
+        pub struct PerRegistryKeyspace<T> {
+            $(pub $ident: T,)*
+        }
+        impl<T> core::ops::Index<RegistryKeyspace> for PerRegistryKeyspace<T> {
```

Comment:
I like that we can use `[RegisterKeyspace]` syntax to access the field. I was not aware that we can do that with any type=)

Reply:
Yep it is nice.